packages/truffle-box/test/box.js

```diff
@@ -195,31 +232,53 @@ describe("truffle-box Box", () => {
       inquirer.prompt.restore();
     });
 
-    describe("when the directory is empty", () => {
-      beforeEach(() => {
-        sinon.stub(regularFs, "readdirSync").returns([]);
+    describe("when directory is empty", () => {
+      before(() => {
+        sinon.stub(fse, "readdirSync").returns([]);
       });
-      afterEach(() => {
-        regularFs.readdirSync.restore();
+      after(() => {
+        fse.readdirSync.restore();
       });
 
-      it("doesn't prompt the user", async () => {
+      it("doesn't prompt user", async () => {
         await Box.checkDir();
         assert.strictEqual(inquirer.prompt.called, false);
       });
     });
 
-    describe("when the directory is non-empty", () => {
-      beforeEach(() => {
-        sinon.stub(regularFs, "readdirSync").returns(["someCrappyFile.js"]);
+    describe("when directory is non-empty", () => {
+      before(() => {
+        sinon.stub(fse, "readdirSync").returns(["someCrappyFile.js"]);
       });
-      afterEach(() => {
-        regularFs.readdirSync.restore();
+      after(() => {
+        fse.readdirSync.restore();
+      });
+
+      it("prompts user", () => {
+        Box.checkDir(options);
+        assert(inquirer.prompt.called);
+      });
+    });
+
+    describe("when directory is non-empty and user declines to unbox", () => {
+      before(() => {
+        sinon.stub(fse, "readdirSync").returns(["someCrappyFile.js"]);
+        sinon.stub(process, "exit").returns(1);
+      });
+      after(() => {
+        fse.readdirSync.restore();
+        process.exit.restore();
       });
 
-      it("prompts the user", () => {
-        Box.checkDir();
+      it("Exits unbox process", async () => {
+        inquirer.prompt.restore();
+        sinon
+          .stub(inquirer, "prompt")
+          .returns(Promise.resolve({ proceed: false }));
+        await Box.checkDir(options);
         assert(inquirer.prompt.called);
+        assert(options.logger.loggedStuff.includes("Unbox cancelled"));
+        assert(process.exit.called);
```

Comment:
Seems wise to add a check that unboxing doesn't actually happen according to `fse`